packages/x-date-pickers-pro/src/DateRangeCalendar/DateRangeCalendar.tsx

```diff
@@ -469,11 +470,18 @@ const DateRangeCalendar = React.forwardRef(function DateRangeCalendar(
   });
 
   const handleDayMouseEnter = useEventCallback(
-    (event: React.MouseEvent<HTMLDivElement>, newPreviewRequest: PickerValidDate) => {
-      if (!isWithinRange(utils, newPreviewRequest, valueDayRange)) {
-        setRangePreviewDay(newPreviewRequest);
+    (event: React.MouseEvent<HTMLDivElement>, newRangePreviewDay: PickerValidDate) => {
```

Comment:
There is one scenario where this does not work: if you hover a day while `value === [null, null]`, you don't move your mouse and you have an update of the value (from some server update).

Before this PR, the preview range would be applied.
Now it would not.

I think the trade-off is fine here, especially given than before this PR is would break already if you hovered inside the range and the value updates (since we do `setRangePreviewDay(null)` on line 476).
I could store the hovered date in a ref if it's not causing any preview and check when the value change, but tbh it feels like wasted complexity.